tikv-client/src/main/java/com/pingcap/tikv/expression/ArithmeticBinaryExpression.java

```diff
@@ -34,38 +35,41 @@
   }
 
   public static ArithmeticBinaryExpression plus(Expression left, Expression right) {
-    return new ArithmeticBinaryExpression(PLUS, left, right);
+    return new ArithmeticBinaryExpression(left.dataType, PLUS, left, right);
```

Comment:
I think we should consider type promotion

Reply:
dataType will be reset in `BasicExpression`.

Reply:
So any data type is valid here? Where is the logic of reset?